source/brailleDisplayDrivers/albatross.py

```diff
@@ -0,0 +1,353 @@
+			self._dev.close()
+			self._dev = None
+			if not self._chkPort(self._currentPort):
+				# Maybe display was unplugged/powered off which causes USB serial port disappearing.
+				self._tryReconnect = True
+				if self._dev:
+					self._dev.close()
+			return False
+		if data == ESTABLISHED:
+			# Actually input buffer should be reseted, but poor man's solution now.
+			while len(self._dev.read(INPUT_BUF_SIZE)):
+				pass
+		return True
+
+	def _keepConnected(self):
+		# Can disappeared port be found again?
+		if self._tryReconnect:
+			if not self._chkPort(self._currentPort):
+				return
+			else:
+				# Port found.
+				self._tryReconnect = False
+				return
+		if self._timerRunning and self.numCells:
+			self._sendToDisplay(BOTH_BYTES)
+
+	def _onReceive(self, data: bytes):
+		if not self.numCells:
+			# If no connection, Albatross sends continuously byte \xff
+			# followed by byte containing various settings like number of cells.
+			# But there may be garbage before \xff.
+			if data != b"\xff":
+				while data != b"\xff" and len(data):
+					data = self._dev.read(1)
+				if not len(data):
+					log.debugWarning("No init byte")
+					return
+			log.debugWarning("Init byte: %r" % data)
+			data = self._dev.read(1)
+			if not len(data):
+				log.debugWarning("No value byte")
+				return
+			log.debugWarning("Value byte: %r" % data)
+			if not self._sendToDisplay(ESTABLISHED):
+				return
+			self.numCells = 80 if ord(data) >> 7 == 1 else 46
+		# Connected.
+		else:
+			# It is possible that there is no connection from perspective of display.
+			if data == b"\xff":
+				if self._sendToDisplay(ESTABLISHED):
+					self._clearOldCells()
+				log.debugWarning("Byte %r, numCells %d, _tryReconnect %r" % (data, self.numCells, self._tryReconnect))
+				return
+			# If Ctrl-key is pressed, then there is at least one byte to read;
+			# in single ctrl-key presses and key combinations the first key is resent as last one.
+			if ord(data) in CONTROL_KEY_CODES:
+				# at most 4 keys.
+				data += self._dev.read(4)
+			pressedKeys = set(data)
+			try:
+				inputCore.manager.executeGesture(InputGestureKeys(pressedKeys))
+			except inputCore.NoInputGestureAction:
+				pass
+
+	def display(self, cells: List[int]):
+		if not self.numCells:
+			return
+		writeBytes: List[bytes] = [START_BYTE, ]
+		# Only changed content is sent (cell index and data).
+		for i, cell in enumerate(cells):
+			if cell != self._oldCells[i]:
+				self._oldCells[i] = cell
+				# display indexing starts from 1
+				writeBytes.append(intToByte(i + 1))
+				# Bits have to be reversed.
+				# Source: https://stackoverflow.com/questions/12681945/reversing-bits-of-python-integer
+				writeBytes.append(intToByte(int('{:08b}'.format(cell)[::-1], 2)))
+		writeBytes.append(END_BYTE)
+		self._sendToDisplay(b"".join(writeBytes))
+
+	gestureMap = inputCore.GlobalGestureMap({
+		"globalCommands.GlobalCommands": {
+			"review_top": ("br(albatross):home1", "br(albatross):home2",),

```

Comment:
Since these names (E.G. `home2`) must match the values in the dictionary, it would be better to use those directly.
If will be more obvious if there is a key enum and keyToName dict:
```python
"review_top": (f"br(albatross):{KeyToName{Keys.HOME_1}", f"br(albatross):{KeyToName{Keys.HOME_2}",),
```

Reply:
Unfortunately it seems not to increase ease of maintenance and/or reading but would require more code. Current aproach is used in NVDA's other braille display drivers as well.